app/src/main/java/com/example/feelgoodinc/fragments/UserProfileFragment.java

```diff
@@ -146,25 +147,35 @@ public void submitNewPassword(View v){
                 @Override
                 public void onSuccess(User user) {
                     // if task succeeds
-                    Toast.makeText(getContext(),
-                            "Password changed", Toast.LENGTH_LONG).show();
-
                     // hide form now that password has been changed
                     View passwordForm = v.findViewById(R.id.passwordForm);
                     passwordForm.setVisibility(View.INVISIBLE);
+                    passwordRequirements.setVisibility(View.INVISIBLE);
+
+                    Toast.makeText(getContext(),
+                            "Password changed", Toast.LENGTH_LONG).show();
                 }
 
                 @Override
                 public void onError(Exception e) {
-                    Log.d("COMP3013", e.toString());
-                    Toast.makeText(getContext(),
-                            "Password could not be changed", Toast.LENGTH_LONG).show();
+                    if (e instanceof IllegalArgumentException && e.getMessage() != null) {
```

Comment:
Same as file before. Perhaps worth making a new interface/exception sub-class to handle this.